core/src/iam/verify.rs

```diff
@@ -157,53 +157,33 @@ pub async fn token(kvs: &Datastore, session: &mut Session, token: &str) -> Resul
 			// Ensure that the transaction is cancelled
 			tx.cancel().await?;
 			// Obtain the configuration to verify the token based on the access method
-			let (au, cf) = match de.kind.clone() {
-				AccessType::Record(at) => {
-					let cf = match at.jwt.verify.clone() {
-						JwtAccessVerify::Key(key) => config(key.alg, key.key.as_bytes()),
-						#[cfg(feature = "jwks")]
-						JwtAccessVerify::Jwks(jwks) => {
-							if let Some(kid) = token_data.header.kid {
-								jwks::config(kvs, &kid, &jwks.url, token_data.header.alg).await
-							} else {
-								Err(Error::MissingTokenHeader("kid".to_string()))
-							}
+			let cf = match de.kind.clone() {
```

Comment:
Do we have to clone each time we test this condition?
Couldn't use a ref, and clone what we need once the condition is met?

Reply:
Good catch! Addressed in [d4d2a4b](https://github.com/surrealdb/surrealdb/pull/4385/commits/d4d2a4bd995b1bdc0a4d7752961755c6b3c72bf2).